deferred_heap.h

```diff
@@ -1041,17 +1048,15 @@ namespace gcpp {
 
 					//	find the end of the allocation
 					auto end_i = i + 1;
-					auto end = pg.page.location_info(pg.page.locations()).pointer;
 					for (; end_i < pg.page.locations(); ++end_i) {
 						auto info = pg.page.location_info(end_i);
 						if (info.is_start) {
-							end = info.pointer;
 							break;
 						}
 					}
 
 					// call the destructors for objects in this range
-					destroy_objects({ start.pointer, end - start.pointer });
+					destroy_objects({ start.pointer, end_i - i });
```

Comment:
Unit mismatch here -- this would be changing from (byte*, #bytes) to (byte*, #locations).